Backend/RecipeIngredientsAlgorithm/Iteration_1/ranking.js

```diff
@@ -0,0 +1,159 @@
+
+const helper = require('./helpers')
+
+///////////////////////////////////   POINTS RULES   //////////////////////////////////////////////
+const OVERLAP_POINTS = 10
+const MEAL_TYPE_POINTS = 3
+const TIME_COOK_POINTS = 8
+const KCAL_POINTS = 8
+
+const TOP_1_2_CUISINE_POINTS = 3
+const TOP_3_4_CUISINE_POINTS = 2
+const TOP_5_CUISINE_POINTS = 1
+const RANDOM_POINTS = 10
+var maxRandomPoints = 0
+var userRandomness = 0
+
+//////////////////////////////////   POINTS    /////////////////////////////////////
+var overlapPoints = 0
+var mealTypePoints = 0
+var timeCookPoints = 0
+var idealKcalPoints = 0
+var cuisinePoints = new Map()
+
+//////////////////////////////// GETTING POINTS FUNCTIONS ////////////////////////////////
+
+// Function to get overlapping elements between two arrays, by dividing the ingredient by words and then storing them into arrays to then combine them and put it in a set to get the non repeating ones.
+// so that in the set are only distinct values and if this value is less than the two arrays combined it means there are words that match, so we count it as overlap of ingredient
+const setOverlapIngredientsPoints = (food, userIngredientsMap) => {
+    const foodIngredients = food.ingredients
+    const overlap = helper.wordSeparation(foodIngredients, userIngredientsMap)
+    const similarPercent = overlap / userIngredientsMap.size
+    overlapPoints = OVERLAP_POINTS * (1 - userRandomness) * similarPercent
+}
+
+const getOverlapIngredientsPoints = () => {
+    return overlapPoints
+}
+
+// Function to get the points of the user based on the hour and the type of meal
+const setHourMealTypePoints = (food, hour) => {
+    if (5 <= hour && hour < 12) {
+        if (food.mealType === 'breakfast') {
+            mealTypePoints = MEAL_TYPE_POINTS
+        }
+    } else if (12 <= hour && hour < 6) {
+        if (food.mealType === 'lunch' || food.mealType === 'lunch/dinner') {
+            mealTypePoints = MEAL_TYPE_POINTS
+        }
+    } else if (6 <= hour && hour <= 23) {
+        if (food.mealType === 'dinner' || food.mealType === 'lunch/dinner') {
+            mealTypePoints = MEAL_TYPE_POINTS
+        }
+    }
+}
+
+const getHourMealTypePoints = () => {
+    return mealTypePoints
+}
+
+// Function to see if the cooking time of a meal satisfies the user input
+const setTimeCookPoints = (food, userTimeCook) => {
+    if (userTimeCook === 'fast' && food.totalTime <= 25) {
+        timeCookPoints = TIME_COOK_POINTS
+    } else if (userTimeCook === 'normal' && (25 < food.totalTime && food.totalTime <= 60)) {
+        timeCookPoints = TIME_COOK_POINTS
+    } else if (userTimeCook === 'long' && (60 < food.totalTime)) {
+        timeCookPoints = TIME_COOK_POINTS
+    }
```

Comment:
since you have a global var, we would need to reset to 0 if none of these were the case. The previous calculation for another recipe could have set timeCookPoints to 8 when in reality it should be 0